presto-parquet/src/test/java/com/facebook/presto/parquet/batchreader/decoders/TestMode.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.batchreader.decoders;
+
+public enum TestMode
```

Comment:
+1 to going with the object-oriented approach. I think just having a superclass with the main logic and maybe delegating the mode-specific details to an abstract method implementation would be better than having to pass in the `TestMode`.

Reply:
As per now I committed an idea integrating both of your suggestions. The `TestMode` is removed. It got replaced by three lightweight methods implemented in three subclasses that refer to the `TestParquetUtils`, where still the main part is handled. Would be curious to know your thoughts, because I still have the feeling it might be not as clean as it can be.